lib/network/tsnetwork_buf.c

```diff
@@ -35,7 +38,7 @@ struct network_buf_cookie {
 	size_t bufpos;
 	struct timeval timeout;
 	struct timeval timeout_max;
-	ssize_t (* sendrecv)(int, void *, size_t, int);
```

Comment:
I'm fairly confident that this is a good change, but I suspect that my commit message is incorrect.

clang-16 definitely refuses to compile the previous version, claiming that it's a function cast between incompatible function types.  As such, replacing the function pointer with an int seems like a good change.

However, c99 6.7.5.3 para 15 contains:

> (In the determination of type compatibility and of a composite type, each parameter declared with function or array type is taken as having the adjusted type and each parameter declared with qualiﬁed type
is taken as having the unqualiﬁed version of its declared type.)

If I'm understanding that correctly, then

```
const f1(char *);
const f2(const char *);
```
should be compatible; in which case, `send()` and `recv()` should be compatible.

Reply:
I think the code as written was correct; as you note, the extra `const` should not affect function type compatibility.  But it's probably a good change anyway since using a flag will probably generate better code than indirecting via a function pointer.

Maybe just change the commit message to a vaguer "some compilers object to using a function pointer this way".